service/frontend/wrappers/ratelimited/api_generated.go

```diff
@@ -0,0 +1,417 @@
+
+// Code generated by gowrap. DO NOT EDIT.
+// template: ../../templates/ratelimited.tmpl
+// gowrap: http://github.com/hexdigest/gowrap
+
+import (
+	"context"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/cache"
+	"github.com/uber/cadence/common/quotas"
+	"github.com/uber/cadence/common/types"
+	"github.com/uber/cadence/service/frontend/api"
+)
+
+// apiHandler implements api.Handler interface instrumented with rate limiter.
+type apiHandler struct {
+	wrapped               api.Handler
+	tokenSerializer       common.TaskTokenSerializer
+	domainCache           cache.DomainCache
+	userRateLimiter       quotas.Policy
+	workerRateLimiter     quotas.Policy
+	visibilityRateLimiter quotas.Policy
+}
+
+// NewAPIHandler creates a new instance of Handler with ratelimiter.
+func NewAPIHandler(
+	wrapped api.Handler,
+	domainCache cache.DomainCache,
+	userRateLimiter quotas.Policy,
+	workerRateLimiter quotas.Policy,
+	visibilityRateLimiter quotas.Policy,
+) api.Handler {
+	return &apiHandler{
+		wrapped:               wrapped,
+		tokenSerializer:       common.NewJSONTaskTokenSerializer(),
+		domainCache:           domainCache,
+		userRateLimiter:       userRateLimiter,
+		workerRateLimiter:     workerRateLimiter,
+		visibilityRateLimiter: visibilityRateLimiter,
+	}
+}
+
+func (h *apiHandler) CountWorkflowExecutions(ctx context.Context, cp1 *types.CountWorkflowExecutionsRequest) (cp2 *types.CountWorkflowExecutionsResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeVisibility, cp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.CountWorkflowExecutions(ctx, cp1)
+}
+
+func (h *apiHandler) DeprecateDomain(ctx context.Context, dp1 *types.DeprecateDomainRequest) (err error) {
+	return h.wrapped.DeprecateDomain(ctx, dp1)
+}
+
+func (h *apiHandler) DescribeDomain(ctx context.Context, dp1 *types.DescribeDomainRequest) (dp2 *types.DescribeDomainResponse, err error) {
+	return h.wrapped.DescribeDomain(ctx, dp1)
+}
+
+func (h *apiHandler) DescribeTaskList(ctx context.Context, dp1 *types.DescribeTaskListRequest) (dp2 *types.DescribeTaskListResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeUser, dp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.DescribeTaskList(ctx, dp1)
+}
+
+func (h *apiHandler) DescribeWorkflowExecution(ctx context.Context, dp1 *types.DescribeWorkflowExecutionRequest) (dp2 *types.DescribeWorkflowExecutionResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeUser, dp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.DescribeWorkflowExecution(ctx, dp1)
+}
+
+func (h *apiHandler) GetClusterInfo(ctx context.Context) (cp1 *types.ClusterInfo, err error) {
+	return h.wrapped.GetClusterInfo(ctx)
```

Comment:
previously user-limited, but this seems fine.  it's a trivial endpoint / other cluster-wide things don't have limits and nobody really calls them anyway.